hr_expense_employee_analytic_default/models/hr_expense.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2024 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class HrExpense(models.Model):
+    _inherit = "hr.expense"
+
+    @api.onchange("product_id", "date", "account_id")
+    def _onchange_product_id_date_account_id(self):
+        """Set a specific context with the employee of the expense."""
+        _self = self.with_context(hr_expense_employee_id=self.employee_id.id)
+        return super(HrExpense, _self)._onchange_product_id_date_account_id()
```

Comment:
No need to put an extra variable, as `self` is a local one:

```suggestion
        self = self.with_context(hr_expense_employee_id=self.employee_id.id)
        return super()._onchange_product_id_date_account_id()
```